### PR TITLE
chore: Use github actions for tests

### PR DESCRIPTION
**Pull request by attiks**
Refs: #OPS-9006

### CODE REVIEW

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Setup PHP with PECL extension
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.0
```
Comment (orakili) [inline .github/workflows/run-tests.yml:13]: The D10 starterkit uses the php8.1 image, should this be 8.1 as well for consistency notably for the `code lint` step.

File: composer.json [hunk 2]
```diff
@@ -69,7 +69,7 @@
         "phpspec/prophecy-phpunit": "^2",
         "phpunit/php-code-coverage": "^9.2",
         "phpunit/phpunit": "^9.5",
-        "weitzman/drupal-test-traits": "^1.5"
+        "weitzman/drupal-test-traits": "^2.0@RC"
```
Comment (orakili) [inline composer.json:72]: Looks like there is already a 2.0.1 release: https://gitlab.com/weitzman/drupal-test-traits/-/tags/2.0.1.

File: phpunit.xml [hunk 3]
```diff
@@ -18,10 +18,10 @@
     <!-- Do not limit the amount of memory tests take to run. -->
     <ini name="memory_limit" value="-1"/>
     <env name="SYMFONY_DEPRECATIONS_HELPER" value="weak"/>
-    <env name="DTT_BASE_URL" value="http://starterkit.test"/>
+    <env name="DTT_BASE_URL" value="http://hrinfo-site.docksal.site/"/>
```
Comment (orakili) [inline phpunit.xml:21]: Leftover from local testing?

File: phpunit.xml [hunk 4]
```diff
@@ -18,10 +18,10 @@
...
     <env name="DTT_API_URL" value="http://localhost:9222"/>
     <!-- Example SIMPLETEST_BASE_URL value: http://localhost -->
-    <env name="SIMPLETEST_BASE_URL" value="http://starterkit.test"/>
+    <env name="SIMPLETEST_BASE_URL" value="http://hrinfo-site.docksal.site/"/>
```
Comment (orakili) [inline phpunit.xml:24]: Leftover from local testing?

Comment (attiks) [reply phpunit.xml]: Copy/paste

**Review by orakili — APPROVED** [review]

**Pull request merged**